liblangutil/Scanner.cpp

```diff
@@ -789,13 +798,17 @@ Token Scanner::scanString()
 				return setError(ScannerError::IllegalEscapeSequence);
 		}
 		else
+		{
+			if (static_cast<unsigned>(c) > 0x7f)
+				return setError(ScannerError::IllegalCharacterInString);
```

Comment:
I still don't like this approach, but oh well.

Reply:
I think we should also disallow some of the control sequences. I would even disallow tabs.

Reply:
If we want to reduce to printable characters that would still include tabs, iirc.

Reply:
You could use `isprint` or check for `ch >= 0x1f && ch <= 0x7f` (7-bit ascii without control character range). I am not sure, is there a usecase to disallow control chars, such as LF or HT?

Reply:
`ch > 0x1f && ch < 0x7f` sounds good to me.

Reply:
Done.